src/charm.py

```diff
@@ -160,6 +161,26 @@ def _configure(self, _):
 
         self.unit.status = ActiveStatus()
 
+    def _update_status(self, _):
+        """Update the status of the charm."""
+        # Run configure to reset the status
+        self._configure(_)
```

Comment:
I am surprised by the need to run configure in update status. That the purpose is to "reset the status". But isn't that the whole purpose of `_update_status()`. Making the Prometheus charm reload its configuration every time an update status event triggers seems unnecessary.

Reply:
`_configure` might be needed here for https://github.com/canonical/prometheus-k8s-operator/issues/262

Reply:
If a health check fails then becomes healthy again, we want to change back to active state. Without _configure we have no way to do that

Reply:
Prometheus status may be set to blocked status if it obtains an invalid configuration from the command line or through relation data. 

Is it not possible for the update status event handler to check that Prometheus is not currently in blocked status due to these reasons and only then set active status if Prometheus is healthy and ready ?

Invoking configure every time in update status implies every time an update status event fires Prometheus will reload its configuration. Could this be a problem ?

Reply:
As @balbirthomas said: Why don't we actually... check whether #262 has happened or what the state is/should be before blindly calling `_configure`?